### PR TITLE
DAH-1158 lottery search results

### PR DESCRIPTION
**Pull request by christine-sfg**
DAH-1158

Implement lottery search results
-Add form submit button
-Add error handling
-Make new api call to get lottery results info
-When lottery number is found, display the ranking info
-When dismissing and reopening the modal, search form is in its original state with the list of preferences for the listing
-Add Lottery Summary to the ListingDetailsLottery component
-Added component level ErrorBoundary

Note
-styling will be polished in a subsequent ticket
-some text is missing and will be added in subsequent tickets for bloom changes and phrase translations

Error Handling
image

Successful Search Results
image

ListingDetails Lottery component with Lottery Summary
image

Component level ErrorBoundary (in case there are issues calling the lottery_buckets api)
image

### CODE REVIEW

File: app/javascript/api/apiEndpoints.ts
Comment (christine-sfg) [inline app/javascript/api/apiEndpoints.ts:5]: i was hoping to use [msw](https://mswjs.io/) to mock up network calls, but looks like we need to update some libs first

File: app/javascript/components/ErrorBoundary.tsx
Comment (jacksongonzales) [inline app/javascript/components/ErrorBoundary.tsx:65]: should this be `case BoundaryScope.page`?

File: app/javascript/modules/listingDetailsAside/ListingDetailsAside.tsx
Comment (jacksongonzales) [inline app/javascript/modules/listingDetailsAside/ListingDetailsAside.tsx:32]: we were investigating [expiring TODOs](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md) recently — I wonder if it'd be possible to do something similar with JIRA tickets 🤔

File: app/javascript/modules/listingDetailsLottery/ListingDetailsLottery.tsx
Comment (jacksongonzales) [inline app/javascript/modules/listingDetailsLottery/ListingDetailsLottery.tsx:62]: what's the reason for moving the `Modal` out of `ListingDetailsLotteryModal`?

File: app/javascript/modules/listingDetailsAside/ListingDetailsAside.tsx
Comment (christine-sfg) [reply app/javascript/modules/listingDetailsAside/ListingDetailsAside.tsx:32]: Luckily this jira is in the sprint so it should get resolved soon, but that's an excellent idea. Perhaps add the jira with an expiring TODO

**Comment by jacksongonzales** [comment]
this looks great! I'll hold off on approving till we can wrangle tests, and lmk if you want another set of eyes on it

File: app/javascript/modules/listingDetailsLottery/ListingDetailsLottery.tsx
Comment (christine-sfg) [reply app/javascript/modules/listingDetailsLottery/ListingDetailsLottery.tsx:62]: There's a lot of dynamic content in the modal content so this made it easier to test without having to keep opening the modal. Perhaps the component in the modal would be better named as `ListingDetailsLotterySearchForm` or something like that?

Comment (jacksongonzales) [reply app/javascript/modules/listingDetailsLottery/ListingDetailsLottery.tsx:62]: ah, got it! I like the idea of renaming the inner component

**Review by jacksongonzales — APPROVED** [review]

**Pull request merged**
